packages/flutter/lib/src/rendering/semantics.dart

```diff
@@ -0,0 +1,456 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+import 'dart:ui' show Rect;
+
+import 'package:flutter/painting.dart';
+import 'package:sky_services/semantics/semantics.mojom.dart' as engine;
+import 'package:vector_math/vector_math_64.dart';
+
+import 'node.dart';
+
+typedef void SemanticAnnotator(SemanticsNode semantics);
+
+/// Interface for RenderObjects to implement when they want to support
+/// being tapped, etc.
+///
+/// These handlers will only be called if the relevant flag is set
+/// (e.g. handleSemanticTap() will only be called if canBeTapped is
+/// true, handleSemanticScrollDown() will only be called if
+/// canBeScrolledVertically is true, etc).
+abstract class SemanticActionHandler {
+  void handleSemanticTap() { }
+  void handleSemanticLongPress() { }
+  void handleSemanticScrollLeft() { }
+  void handleSemanticScrollRight() { }
+  void handleSemanticScrollUp() { }
+  void handleSemanticScrollDown() { }
+}
+
+enum _SemanticFlags {
+  mergeAllDescendantsIntoThisNode,
+  canBeTapped,
+  canBeLongPressed,
+  canBeScrolledHorizontally,
+  canBeScrolledVertically,
+  hasCheckedState,
+  isChecked,
+}
+
+/// A BitField over an enum (or other class whose values implement "index").
+/// Only the first 63 values of the enum can be used as indices.
+class BitField<T extends dynamic> {
```

Comment:
Please move this class into its own file.  We should probably move it into another package, but for now a separate file is fine.